src/vconsole/vconsole-setup.c

```diff
@@ -521,13 +521,19 @@ static int find_source_vc(char **ret_path, unsigned *ret_idx) {
         assert(ret_path);
         assert(ret_idx);
 
+        /* This function returns -EBUSY only when it finds VCs and all are busy. If one of them can't be used
+         * for any other reason, the corresponding error is returned instead. It returns -ENOENT when no VC
+         * is found at all. */
+
         for (unsigned i = 1; i <= 63; i++) {
                 _cleanup_close_ int fd = -EBADF;
                 _cleanup_free_ char *path = NULL;
 
                 r = verify_vc_allocation(i);
                 if (r < 0) {
                         log_debug_errno(r, "VC %u existence check failed, skipping: %m", i);
+                        if (r != -ENOENT && err == -EBUSY)
+                                err = 0; /* EBUSY has less priority than any other error except for ENOENT. */
```

Comment:
this is a weird way to express. just handle the r == -EBUSY case separately, and set a boolean "ebusy" or so, and then when we are done here either return r, or return EBUSY in case the "ebusy" bool is set.

Reply:
We want errors occurring when opening the device have priority over any other errors and "no VC found at all" the lowest priority. I added a comment to explain this.